# Gamezone Exploratory Data Analysis

### Connect to the Gamezone DB, pull the data, clean, and commit to Github 

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
import plotly.graph_objects as go

Connect to the Gamezone DB, pull the data, clean, and commit to Github 

In [4]:
#%pip install mysqlclient
from sqlalchemy import create_engine
import mysql.connector


# Create database connection for MySQL
engine = create_engine('mysql+mysqldb://root:fst6Just_Data@localhost:3306/gamezone')

# Read orders table into a dataframe
orders_df = pd.read_sql_query('SELECT * FROM orders', engine)

# Display first few rows
print(orders_df.head(10))

    user_id           order_id purchase_date   ship_date  \
0  2c06175e   0001328c3c220830    2020-12-24  2020-12-13   
1  ee8e5bc2  0002af7a5c6100772    2020-10-01  2020-09-21   
2  9eb4efe0   0002b8350e167074    2020-04-21  2020-02-16   
3  cac7cbaf   0006d06b98385729    2020-04-07  2020-04-04   
4  6b0230bc   00097279a2f46150    2020-11-24  2020-08-02   
5  046e7528    000f169e17f4941    2020-08-20  2020-08-09   
6  94dfec76   000fdd9bfe463101    2020-05-11  2020-03-05   
7  8a328a1d   0010d31e3c058676    2020-07-03  2020-04-03   
8  46c57b58   0013ac51a8831707    2019-03-24  2018-10-29   
9  a2eaa1d1   001620ea42468912    2019-11-01  2019-10-17   

                     product_name    price purchasing_platform  \
0                 Nintendo Switch   168.00             website   
1                 Nintendo Switch   160.61             website   
2                 Nintendo Switch   151.20             website   
3       Sony PlayStation 5 Bundle  1132.82             website   
4        

### Initial Inspection

In [5]:
display(orders_df.head(10))

user_id           order_id purchase_date   ship_date  \
0  2c06175e   0001328c3c220830    2020-12-24  2020-12-13   
1  ee8e5bc2  0002af7a5c6100772    2020-10-01  2020-09-21   
2  9eb4efe0   0002b8350e167074    2020-04-21  2020-02-16   
3  cac7cbaf   0006d06b98385729    2020-04-07  2020-04-04   
4  6b0230bc   00097279a2f46150    2020-11-24  2020-08-02   
5  046e7528    000f169e17f4941    2020-08-20  2020-08-09   
6  94dfec76   000fdd9bfe463101    2020-05-11  2020-03-05   
7  8a328a1d   0010d31e3c058676    2020-07-03  2020-04-03   
8  46c57b58   0013ac51a8831707    2019-03-24  2018-10-29   
9  a2eaa1d1   001620ea42468912    2019-11-01  2019-10-17   

                     product_name    price purchasing_platform  \
0                 Nintendo Switch   168.00             website   
1                 Nintendo Switch   160.61             website   
2                 Nintendo Switch   151.20             website   
3       Sony PlayStation 5 Bundle  1132.82             website   
4                 Nintendo Switch    33.89             website   
5                 Nintendo Switch   168.00             website   
6          27in 4K gaming monitor   390.59             website   
7          27in 4K gaming monitor   480.00             website   
8  JBL Quantum 100 Gaming Headset    22.22             website   
9                 Nintendo Switch   168.00             website   

  marketing_channel  account country_code  
0         affiliate  unknown           US  
1            direct  desktop           DE  
2            direct  desktop           US  
3            direct  desktop           AU  
4            direct  desktop           TR  
5            direct  desktop           PH  
6            direct  desktop           GB  
7            direct  desktop           BR  
8            direct  desktop           BE  
9            direct  desktop           US

In [6]:
display(orders_df.tail(10))

user_id           order_id purchase_date   ship_date  \
21843  72b4c703   ffecb455fb849256    2020-08-10  2020-08-13   
21844  b50f2404   ffed03b71ea76708    2019-06-04  2019-06-06   
21845  7908c639   ffeec4c209751749    2021-01-20  2021-01-23   
21846  7908c639   ffeec4c209751750    2021-01-20  2021-01-23   
21847  6bd1f1a6   ffef00f005b46517    2019-11-21  2019-11-24   
21848  e432cf6f   fff0b6a1e9996384    2019-11-27  2019-11-29   
21849  f4f11c04  fff4592dc6d103535    2019-12-23  2019-12-26   
21850  f4f11c04  fff4592dc6d103537    2019-12-23  2019-12-26   
21851  203ce4dd   fff829b061e16171    2021-01-31  2021-02-02   
21852  203ce4dd   fff829b061e16172    2021-01-31  2021-02-02   

                    product_name    price purchasing_platform  \
21843            Nintendo Switch   168.00             website   
21844            Nintendo Switch   142.80             website   
21845    Lenovo IdeaPad Gaming 3   599.00             website   
21846    Lenovo IdeaPad Gaming 3   599.00             website   
21847            Nintendo Switch   136.95             website   
21848  Sony PlayStation 5 Bundle  1527.06             website   
21849     27in 4K gaming monitor   467.88             website   
21850     27in 4K gaming monitor   467.88             website   
21851          Dell Gaming Mouse    67.07          mobile app   
21852          Dell Gaming Mouse    67.07          mobile app   

      marketing_channel  account country_code  
21843             email  desktop           US  
21844      social media   tablet           US  
21845            direct  desktop           US  
21846            direct  desktop           US  
21847            direct  desktop           SE  
21848            direct  desktop           US  
21849            direct  desktop           US  
21850            direct  desktop           US  
21851            direct  desktop           GB  
21852            direct  desktop           GB

In [ ]:
# Check the shape of the dataframe
print("Shape of Orders df")
display(orders_df.shape)

#Check summary stats
print("Summary stats of Orders df")
display(orders_df.describe(include='all'))

Shape of Orders df


(21853, 10)

Summary stats of Orders df


user_id          order_id purchase_date   ship_date     product_name  \
count      21853             21853         21853       21853            21853   
unique     19840             21708           772         825                9   
top     07868dba  fecc9d6766231653    2020-12-19  2020-12-20  Nintendo Switch   
freq           4                 2            95          81            10385   
mean         NaN               NaN           NaN         NaN              NaN   
std          NaN               NaN           NaN         NaN              NaN   
min          NaN               NaN           NaN         NaN              NaN   
25%          NaN               NaN           NaN         NaN              NaN   
50%          NaN               NaN           NaN         NaN              NaN   
75%          NaN               NaN           NaN         NaN              NaN   
max          NaN               NaN           NaN         NaN              NaN   

               price purchasing_platform marketing_channel  account  \
count   21848.000000               21853             21770    21770   
unique           NaN                   2                 5        5   
top              NaN             website            direct  desktop   
freq             NaN               19773             17423    16449   
mean      281.412964                 NaN               NaN      NaN   
std       366.209801                 NaN               NaN      NaN   
min         0.000000                 NaN               NaN      NaN   
25%       126.000000                 NaN               NaN      NaN   
50%       168.000000                 NaN               NaN      NaN   
75%       357.000000                 NaN               NaN      NaN   
max      3146.880000                 NaN               NaN      NaN   

       country_code  
count         21815  
unique          150  
top              US  
freq          10289  
mean            NaN  
std             NaN  
min             NaN  
25%             NaN  
50%             NaN  
75%             NaN  
max             NaN

In [11]:
# Check the data types of the columns
print("Data types of Orders df")
display(orders_df.dtypes)

# Info about the dataframe
print("Info about Orders df")
display(orders_df.info())

Data types of Orders df


user_id                 object
order_id                object
purchase_date           object
ship_date               object
product_name            object
price                  float64
purchasing_platform     object
marketing_channel       object
account                 object
country_code            object
dtype: object

Info about Orders df
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21853 entries, 0 to 21852
Data columns (total 10 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   user_id              21853 non-null  object 
 1   order_id             21853 non-null  object 
 2   purchase_date        21853 non-null  object 
 3   ship_date            21853 non-null  object 
 4   product_name         21853 non-null  object 
 5   price                21848 non-null  float64
 6   purchasing_platform  21853 non-null  object 
 7   marketing_channel    21770 non-null  object 
 8   account              21770 non-null  object 
 9   country_code         21815 non-null  object 
dtypes: float64(1), object(9)
memory usage: 1.7+ MB


None

In [ ]:
# Check for missing values
print("Missing values in Orders df")
missing_values = orders_df.isnull().sum()
missing_values = missing_values[missing_values > 0]

# Dataframe to show missing values and their percentage
missing_info = pd.DataFrame({
    'Missing Values': missing_values,
    'Percentage': (missing_values / len(orders_df)) * 100
})

missing_percentage = (missing_values / len(orders_df)) * 100
# missing_values = pd.DataFrame({'Missing Values': missing_values, 'Percentage': missing_percentage})


display(missing_values)



Missing values in Orders df


price                 5
marketing_channel    83
account              83
country_code         38
dtype: int64

### Notes on data cleansing

Info about Orders df
Data columns (total 10 columns):
 ####   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   user_id:              21853 non-null  object | 
 1   order_id:             21853 non-null  object | 
 2   purchase_date:        21853 non-null  object | 
 3   ship_date:            21853 non-null  object | 
 4   product_name:         21853 non-null  object | 
 5   price:                21848 non-null  float64 |
 6   purchasing_platform:  21853 non-null  object | 
 7   marketing_channel:    21770 non-null  object | 
 8   account:              21770 non-null  object | 
 9   country_code:         21815 non-null  object | 
dtypes: float64(1), object(9)



### Missing values in Orders df

price:                 5 |
marketing_channel:    83 |
account:              83 |
country_code:         38 |
dtype: int64

### Summary Statistics Analysis

Let's break down the key insights from the summary statistics:

1. **Price Statistics**:
   - Analyze min, max, mean, and median prices
   - Check for potential outliers
   - Understanding price distribution

2. **Date Range Analysis**:
   - Review order date range
   - Analyze shipping patterns

3. **Platform & Marketing Distribution**:
   - Check most common platforms
   - Analyze marketing channel effectiveness

4. **Geographic Distribution**:
   - Count orders by country
   - Analyze regional patterns

In [8]:
# Price Analysis
print("\nPrice Statistics:")
print(f"Average Price: ${orders_df['price'].mean():.2f}")
print(f"Median Price: ${orders_df['price'].median():.2f}")
print(f"Price Range: ${orders_df['price'].min():.2f} - ${orders_df['price'].max():.2f}")

# Date Range Analysis
print("\nDate Range:")
print(f"First Order: {orders_df['purchase_date'].min()}")
print(f"Last Order: {orders_df['purchase_date'].max()}")

# Platform Distribution
print("\nTop 5 Purchase Platforms:")
print(orders_df['purchasing_platform'].value_counts().head())

# Marketing Channel Analysis
print("\nMarketing Channel Distribution:")
print(orders_df['marketing_channel'].value_counts())

# Geographic Distribution
print("\nTop 10 Countries by Orders:")
print(orders_df['country_code'].value_counts().head(10))


Price Statistics:
Average Price: $281.41
Median Price: $168.00
Price Range: $0.00 - $3146.88

Date Range:
First Order: 2019-01-01
Last Order: 2021-02-28

Top 5 Purchase Platforms:
purchasing_platform
website       19773
mobile app     2080
Name: count, dtype: int64

Marketing Channel Distribution:
marketing_channel
direct          17423
email            3256
affiliate         721
social media      323
unknown            47
Name: count, dtype: int64

Top 10 Countries by Orders:
country_code
US    10289
GB     1807
CA      956
AU      893
DE      855
FR      558
JP      524
BR      494
ES      407
NL      326
Name: count, dtype: int64
